### ISSUE TITLE
Help: Navigation to a top-level page from deeply nested page

### ISSUE THREAD

**Issue by QuirijnGB** [Milad-Akarie/auto_route_library#442]
Heya, 
First off, thanks for the amazing library!

I've been struggling to find the correct way to navigate from the `HomePage` to the `ProfilePage`. I can seem to get a hold of the correct router.

```dart
@MaterialAutoRouter(
  replaceInRouteName: 'Page,Route,Screen',
  preferRelativeImports: false,
  generateNavigationHelperExtension: true,
  routes: <AutoRoute>[
    CustomRoute(
      transitionsBuilder: TransitionsBuilders.fadeIn,
      page: RoutingScreen,
      name: 'RoutingScreenRouter',
      initial: true,
      children: [
        AutoRoute(
          page: AuthenticationShell,
          path: 'authentication',
          children: <AutoRoute>[
            AutoRoute(path: '', page: AuthenticationScreen),
          ],
        ),
        AutoRoute(path: 'profile', page: ProfilePage),
        CustomRoute(
          transitionsBuilder: TransitionsBuilders.fadeIn,
          path: '',
          page: MainScreen,
          name: 'MainScreenRouter',
          guards: [AuthGuard],
          usesTabsRouter: true,
          children: <AutoRoute>[
            AutoRoute(
              path: '',
              name: 'HomePageRouter',
              page: EmptyRouterPage,
              children: <AutoRoute>[
                AutoRoute(path: '', page: HomePage),
                AutoRoute(path: 'details', page: DetailsPage),
              ],
            ),
            AutoRoute(path: 'payments', page: PaymentsPage),
          ],
        ),
      ],
    ),
  ],
)
class $AppRouter {}
```

Thanks <3

**Comment by Milad-Akarie** [comment]
Hey @QuirijnGB That's a lot of nesting, I guess I'm gonna have to come up with a neater way to do this. 
you could go `context.tabsRouter` that will get you the `MainScreenRouter` then you want to call `parent<T>()` to access it's parent which's the stack that contains your `ProfilePage`

` context.tabsRouter.parent<StackRouter>()`

**Comment by QuirijnGB** [comment]
So I tried
```dart
context.tabsRouter.parent<StackRouter>().navigate(const ProfilePageRoute());
```
and still get
```
I/flutter (13733): ══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
I/flutter (13733): The following _CastError was thrown while handling a gesture:
I/flutter (13733): type '<T extends RoutingController>() => T' is not a subtype of type 'StackRouter' in type cast
I/flutter (13733): 
I/flutter (13733): When the exception was thrown, this was the stack:
I/flutter (13733): #0      ParallelBranchEntry.parent (package:auto_route/src/router/controller/routing_controller.dart:211:53)
```

Ive also tried the following: 
```dart
          AutoRouter.of(context) // HomePageRouter
                .parent<TabsRouter>() // MainScreenRouter
                .parent<StackRouter>() // RoutingScreenRouter <-- fails
```
And it results in the same error. For some reason it doesn't like to get the TabRouter's parent.

Ill see if i can post a small project

**Comment by QuirijnGB** [comment]
Sorry for the late response. Im on 1.0.2

**Comment by quirijnAtPropertyMe** [comment]
Fixed by #486 <3
You can close this @Milad-Akarie

**Comment by QuirijnGB** [comment]
Fixed by #486! 
Thanks @Milad-Akarie
